rest/src/main/java/io/syndesis/rest/v1/util/Urls.java

```diff
@@ -44,4 +60,58 @@ public static URI apiBase(final HttpServletRequest httpRequest) {
         return "/" + parts[1] + "/" + parts[2] + "/";
     }
 
+    /* default */ static URI currentUri(final HttpServletRequest httpRequest) {
+        // in development we have Angular CLI proxying to HA proxy (openshift
+        // router) that proxies to the syndesis-rest; we need the origin
+        // hostname/port/scheme/
+        final String origin = httpRequest.getHeader(ORIGIN_HEADER);
```

Comment:
The Angular CLI has a middleware proxy configured through `proxy.conf.js`: https://github.com/syndesisio/syndesis-ui/blob/master/proxy.conf.js#L10 -- it's a header that's set by the development environment for the UI.